File: app/src/main/java/com/example/le_androidapp/OverviewFragment.java
package com.example.le_androidapp;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;

import com.example.le_androidapp.data.ConnectionState;
import com.example.le_androidapp.domain.DeviceViewModel;

import javax.inject.Inject;

import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class OverviewFragment extends Fragment {

    @Inject
    DeviceViewModel deviceViewModel;

    TextView pitchView;
    TextView rollView;
    TextView flexView;

    public OverviewFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_overview, container, false);

        pitchView = (TextView) view.findViewById(R.id.pitchView);
        rollView = (TextView) view.findViewById(R.id.rollView);
        flexView = (TextView) view.findViewById(R.id.flexView);
//
//        deviceViewModel.getConnectionState().observe(getViewLifecycleOwner(), new Observer<ConnectionState>() {
//            @Override
//            public void onChanged(ConnectionState connectionState) {
//                Log.e("OverviewFragment", "LifeCycleOnChanged");
//                if (connectionState instanceof ConnectionState.Connected) {
//                    pitchView.setText(Float.toString(((ConnectionState.Connected) connectionState).getXVal()));
//                    rollView.setText(Float.toString(((ConnectionState.Connected) connectionState).getYVal()));
//                    flexView.setText(Float.toString(((ConnectionState.Connected) connectionState).getZVal()));
//                }
//            }
//        });

        return view;
    }
}

File: app/src/main/java/com/example/le_androidapp/ui/OverViewFragment.kt
package com.example.le_androidapp.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.lifecycleScope
import androidx.lifecycle.repeatOnLifecycle
import com.example.le_androidapp.databinding.FragmentOverviewBinding
import com.example.le_androidapp.domain.ResultViewModel
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.flow.collectLatest
import kotlinx.coroutines.launch

@AndroidEntryPoint
class OverViewFragment : Fragment() {
    private val resultViewModel : ResultViewModel by viewModels()
    private var _binding : FragmentOverviewBinding? = null
    private val binding get() = _binding!!


    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentOverviewBinding.inflate(inflater,container,false)
        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        viewLifecycleOwner.lifecycleScope.launch {
            repeatOnLifecycle(Lifecycle.State.STARTED){
                resultViewModel.result.collectLatest {
                    with(binding){
                        rollValue.text = it?.rollNumber.toString()
                        pitchValue.text = it?.pitchNumber.toString()
                        flexValue.text = it?.flexNumber.toString()
                    }
                }
            }
        }
    }

}